linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -2714,7 +2714,9 @@ private[emitter] class FunctionEmitter(sjsGen: SJSGen) {
 
     def isMaybeHijackedClass(tpe: Type): Boolean = tpe match {
       case ClassType(className) =>
-        MaybeHijackedClasses.contains(className)
+        HijackedClasses.contains(className) ||
+        className != ObjectClass && globalKnowledge.hijackedDescendants(className).nonEmpty
```

Comment:
```suggestion
        className != ObjectClass && globalKnowledge.isAncestorOfHijackedClass(className)
```

Reply:
This code is called for every `Apply` node in a method, and the most common result is the one that has to go through all those tests. Could we precompute the total set (formerly `MaybeHijackedClasses`) at most once?

Reply:
I'm not sure I understand. Do you expect performance improvements if we do a single `Set` lookup rather than this test? We can precompute the set in `KnowledgeGuardian` if you think it is worth it.

Reply:
That's what I would expect, but maybe it's not such a big deal.